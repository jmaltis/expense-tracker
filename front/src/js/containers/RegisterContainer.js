import React, {Component} from "react";
import AccountForm from "../components/AccountForm";

export default class extends Component {
    register = (values) => {
        console.log(values);
        // TODO call backend
    };

    render() {
        return (
            <AccountForm onSubmit={this.register} submitText="Register"/>
        )
    }
}