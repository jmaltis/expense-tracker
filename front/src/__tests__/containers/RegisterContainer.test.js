import React from "react";
import RegisterContainer from "../../js/containers/RegisterContainer";
import renderer from "react-test-renderer";

test('RegisterContainer should be rendered properly', () => {
    const rendered = renderer.create(
        <RegisterContainer />
    );
    expect(rendered.toJSON()).toMatchSnapshot();
});