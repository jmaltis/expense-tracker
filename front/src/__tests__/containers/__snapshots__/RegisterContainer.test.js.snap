// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterContainer should be rendered properly 1`] = `
<form
  className="accountForm"
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <input
      className="form-control"
      id="login"
      name="login"
      onChange={[Function]}
      placeholder="Login"
      type="text"
      value={undefined}
    />
  </div>
  <div
    className="form-group"
  >
    <input
      className="form-control"
      id="password"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value={undefined}
    />
  </div>
  <button
    className="btn btn-primary"
    disabled={false}
    type="submit"
  >
    Register
  </button>
</form>
`;
